From 2da8702bc101c9dc15d083f784e18115992b3b1b Mon Sep 17 00:00:00 2001
From: Christophe Priouzeau <christophe.priouzeau@foss.st.com>
Date: Wed, 27 Sep 2023 12:00:27 +0200
Subject: [PATCH] Build: Add stub functions of system calls

GNU Arm compiler version greater equal than *11.3.Rel1*
has a linker issue that required system calls are missing,
such as _read and _write. Add stub functions of required
system calls to solve this issue.

Signed-off-by: Chendi Sun <chendi.sun@arm.com>
---
 app/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/app/CMakeLists.txt b/app/CMakeLists.txt
index 603fcc81..514d576a 100755
--- a/app/CMakeLists.txt
+++ b/app/CMakeLists.txt
@@ -254,6 +254,8 @@ endif()
 add_library(tfm_test_app STATIC EXCLUDE_FROM_ALL)
 
 target_sources(tfm_test_app
+    PUBLIC
+        $<$<BOOL:${CONFIG_GNU_SYSCALL_STUB_ENABLED}>:${CMAKE_SOURCE_DIR}/platform/ext/common/syscalls_stub.c>
     PRIVATE
         test_app.c
 )
-- 
2.34.1

